### ISSUE TITLE
Removing and documenting old json format dependencies in Simulation

### ISSUE THREAD

**Issue by xiamaz** [PEDIA-Charite/PEDIA-workflow#13]
Currently the simulation depends quite specifically on the oldJson format in a somewhat intransparent manner. It is at first not entirely clear which fields are used under which assumptions.

This can greatly increases the fragility of our codebase to future changes by Face2Gene.

We can ameliorate this issue only through better documentation of the simulation process and its dependencies on the case json files.

The current workflow in 3_simulation depends on a rather complicated processing sequence involving exome simulation, variant annotation using jannovar, simdrom and pedia-simulator (project called exome-simulator). Of these only the pedia-simulator does json annotation and has hard dependencies on the old json format. Since fortunately the codebase it was quite straightforward to identifity the dependencies on the geneList present in the json.

This could be improved in a manner of ways, which should be thoroughly discussed:
* usage of geneList assumed that this information is provided as-is by face2gene and not inferred in a rather imperfect ways from syndromic omim ids. in order to correctly emulate the geneList by Face2Gene we would actually have to obtain the process by which they were generating these lists.
* restructuring the input file format could enable scoring and simulation on a syndromic level rather than the genetic level, since most of our data sources work on the syndromic level, this might be interesting to consider, but would be a rather involved process

Additionally this process seems to facilitate the usage of snakemake, although this might not actually scale well for single case usage as is wanted for the case of DPDL. Moving the snakemake into python calls to java on a binary or even function level is entirely possible and might be useful for unifying the processing process.
Alternatively it should be quite trivial to add the QC process to the simulation snakemake, as we would just need to further extend the json file dependencies backwards to the input json file level.
Personally I find the snakemake process rather clumsy as it encourages large amounts of io processing between separate program parts (cmd args boilerplate needed a lot), thus making stronger integration more difficult. Since most of these program parts are not really self-contained or useful in a general way I do not think the UNIX design philosophy involving small program parts and large amounts of piping really applies here to justify this manner of operation.

Ideally we should define the required input values for the simulation create test cases and use these for easier QC and preprocessing development.

This issue will be used to track the progress in documenting the simulation process in a somewhat clearer manner.

**Comment by la60312** [comment]
1. 
First of all, we can not control the input format from F2G although we have a cooperation. Ideally, they should define the format and inform us what they changed, but it won't work in reality. Moreover, they will change the format totally in near future. They plan to use LAB api instead of amazon server. So, they don't expect they will provide as any thing related to the JSON files in amazon server. They do provide the format documentation in Lab api. However, the LAB api is still in development. 
I would say someone documents the format we used (old format) now. We assume this is the standard format and we also assume our previous results are accurate. We focus on reproduce the results we have before. Therefore, we assume the original format is the standard.

2.
Secondly, the table used for mapping between phenotype and genes is important to be identical with the one in F2G. We will discuss with them how to co-work on this part, Now, we use the table from OMIM. 

3.
Prioritizing on syndromic level is our next step. We can only just keep the suggested_syndrome list. Then, map the gene back to phenotype to add CADD scores. However, it is difficult. The syndrome name from F2G may differ from Phenomizaer, and the mapping from gene to phenotype from F2G is also the problem.

4. 
The simulation involves lots of steps of processing VCF files and JSON files. Snakemake is designed for integrating the different scripts or tools in the same workflow. By defining the path, you can easily work in different workflows in the same repository. It also enables processing in parallel. It help you solve lots of annoying parts. I won't say it is the state of the art tool for bioinformatics pipeline. The most important thing is that it already worked. We have limited resources and time. I don't think it is a good idea to change this part. 

5. 
The current workflow is able to running on single case. I can simply run snakemake xxxxx/21147.out in classifier, then it will automatically run the entire workflow except QC.

In summary, we document our format based on the original JSON format. However, I have to work on new_download branch to let F2G be able to run it. Someone document it first. Then I check it if there is something incorrect.

**Comment by xiamaz** [comment]
#16 adds documentation on the output file format necessary for the simulation. Since this pull request has been approved, this issue should have been resolved.
